File: esp32/receive.c
<?php
$servername = "sql313.infinityfree.com";
$username = "if0_39717260";     
$password = "AgPV5fsm3HK";
$dbname = "if0_39717260_npk_db";

// Connect to database
$conn = new mysqli($servername, $username, $password, $dbname);
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

// Get data from ESP32
$value1 = $_POST['temperature'] ?? '';
$value2 = $_POST['humidity'] ?? '';

// Insert into database
$sql = "INSERT INTO sensor_data (temperature, humidity) VALUES ('$value1', '$value2')";
if ($conn->query($sql) === TRUE) {
    echo "Data stored successfully";
} else {
    echo "Error: " . $conn->error;
}

$conn->close();
?>

File: web/api/receive.php
<?php
include './connection.php';

$userId     = $_POST['userId'] ?? '';   // email
$pin        = $_POST['pin'] ?? '';      // password
$plotId     = $_POST['plotId'] ?? '';
$nitrogen   = $_POST['nitrogen'] ?? '';
$phosphorus = $_POST['phosphorus'] ?? '';
$potassium  = $_POST['potassium'] ?? '';
$moisture   = $_POST['moisture'] ?? '';

$response = ['success' => 0, 'message' => ''];

$stmt = $mysqli->prepare("SELECT id FROM users WHERE email = ? AND password = ? AND is_deleted = 0 LIMIT 1");
$stmt->bind_param("ss", $userId, $pin);
$stmt->execute();
$result = $stmt->get_result();

if ($result && $result->num_rows > 0) {
	$user = $result->fetch_assoc();
	$dbUserId = $user['id'];

	$insert = $mysqli->prepare("INSERT INTO sensor_data (user_id, pin, plot_id, nitrogen, phosphorus, potassium, moisture) 
                                VALUES (?, ?, ?, ?, ?, ?, ?)");
	$insert->bind_param("sssssss", $userId, $pin, $plotId, $nitrogen, $phosphorus, $potassium, $moisture);

	if ($insert->execute()) {
		$response['success'] = 1;
		$response['message'] = "Data stored successfully";
	} else {
		$response['message'] = "Error inserting data: " . $mysqli->error;
	}
	$insert->close();
} else {
	$response['message'] = "Invalid user credentials";
}

$stmt->close();
$mysqli->close();

echo json_encode($response);
